internal/internal_workflow.go

```diff
@@ -1347,3 +1356,59 @@ func (h *queryHandler) execute(input []byte) (result []byte, err error) {
 	}
 	return result, err
 }
+
+// Add adds delta, which may be negative, to the WaitGroup counter.
+// If the counter becomes zero, all goroutines blocked on Wait are released.
+// If the counter goes negative, Add panics.
+//
+// Note that calls with a positive delta that occur when the counter is zero
+// must happen before a Wait. Calls with a negative delta, or calls with a
+// positive delta that start when the counter is greater than zero, may happen
+// at any time.
+// Typically this means the calls to Add should execute before the statement
+// creating the goroutine or other event to be waited for.
+// If a WaitGroup is reused to wait for several independent sets of events,
+// new Add calls must happen after all previous Wait calls have returned.
+//
+// param delta int32 -> the value to increment the WaitGroup counter by
+func (wg *waitGroupImpl) Add(delta int32) {
+	if (wg.waiting && delta > 0) && (wg.n == 0) {
```

Comment:
I don't think it is possible. Wait call is not going to block if n==0 .

Reply:
It's not. Ran into this while writing unit tests. It's not a possible case in workflow context. I'll remove.